src/components/ha-combo-box.ts

```diff
@@ -226,65 +225,73 @@ export class HaComboBox extends LitElement {
   }
 
   private _openedChanged(ev: ComboBoxLightOpenedChangedEvent) {
-    const opened = ev.detail.value;
-    // delay this so we can handle click event before setting _opened
-    setTimeout(() => {
-      this.opened = opened;
-    }, 0);
+    this.opened = ev.detail.value;
     // @ts-ignore
     fireEvent(this, ev.type, ev.detail);
 
-    if (opened) {
-      this.removeInertOnOverlay();
-    }
-  }
-
-  private removeInertOnOverlay() {
-    if ("MutationObserver" in window && !this._overlayMutationObserver) {
+    if (this.opened) {
       const overlay = document.querySelector<HTMLElement>(
         "vaadin-combo-box-overlay"
       );
 
-      if (!overlay) {
-        return;
+      if (overlay) {
+        this._removeInert(overlay);
       }
+      this._observeBody();
+    } else {
+      this._bodyMutationObserver?.disconnect();
+      this._bodyMutationObserver = undefined;
+    }
+  }
 
-      this._overlayMutationObserver = new MutationObserver((mutations) => {
+  private _observeBody() {
+    if ("MutationObserver" in window && !this._bodyMutationObserver) {
+      this._bodyMutationObserver = new MutationObserver((mutations) => {
         mutations.forEach((mutation) => {
-          if (
-            mutation.type === "attributes" &&
-            mutation.attributeName === "inert"
-          ) {
-            this._overlayMutationObserver?.disconnect();
-            this._overlayMutationObserver = undefined;
-            overlay.inert = false;
-          } else if (mutation.type === "childList") {
-            mutation.removedNodes.forEach((node) => {
-              if (node.nodeName === "VAADIN-COMBO-BOX-OVERLAY") {
-                this._overlayMutationObserver?.disconnect();
-                this._overlayMutationObserver = undefined;
-              }
-            });
-          }
+          mutation.addedNodes.forEach((node) => {
+            if (node.nodeName === "VAADIN-COMBO-BOX-OVERLAY") {
+              this._removeInert(node as HTMLElement);
+            }
+          });
+          mutation.removedNodes.forEach((node) => {
+            if (node.nodeName === "VAADIN-COMBO-BOX-OVERLAY") {
+              this._overlayMutationObserver?.disconnect();
+              this._overlayMutationObserver = undefined;
```

Comment:
Shouldn't we stop observing the body to here?

Reply:
No because if the results empty, the overlay is removed. And if the user change the filter string, we should continue to listen in the case the overlay come back (if results are not empty anymore). That's why I only stop the observer when opened = false.